app/controllers/api/functions.php

```diff
@@ -1750,6 +1751,13 @@
                 ->setAttribute('responseStatusCode', 500)
                 ->setAttribute('errors', $th->getMessage() . '\nError Code: ' . $th->getCode());
             Console::error($th->getMessage());
+
+            if ($th instanceof AppwriteException) {
+                if ($function->getAttribute('logging')) {
+                    Authorization::skip(fn () => $dbForProject->createDocument('executions', $execution));
+                }
+                throw $th;
+            }
```

Comment:
@loks0n Can you update this code block with changes we discussed on the other PR?